submariner/multiple-cable-drivers.md

```diff
@@ -0,0 +1,415 @@
+
+    // Topology identifies the connection topology for the cable ends: full-mesh/point-to-point/client-server.
+    Topology `json:"topology,omitempty"`
+
+   // Priority identifies the priority for the connection policy. Higher priorities are applied first.
+    Priority  *int  `json:"priority,omitempty"`
+
+    // ConnectionPolicy identifies the default connection policy if no policies are found: to connect or no-connect.
+    ConnectionPolicy *string `json:"connectionPolicy,omitempty"`
+
+    Spec CableDriverSpec `json:"spec"`
+}
+
+type CableDriverSpec struct {
+    // Name of the cable driver implementation.
+    Name string `json:"name"`
+
+    // Options specifies key/value configuration parameters for the cable driver.
+    Options *v1.ConfigMap `json:"options,omitempty"`
+}
+```
+
+The `ClusterConnectionPolicies` CR defines the criteria for selecting which cluster pairs/subsets can connect and associated optional
+configuration options to use for each cluster connection. To resolve the connectivity policy for a remote cluster's Endpoint,
+Submariner would:  
+
+- Search for a policy whose left and right label selectors match the corresponding Cluster resources of the local
+and remote Endpoints.
+- Search for a policy whose cluster label selector match the corresponding Cluster resources of the local and remote Endpoints.
+
+If any matching policy is found, the clusters are either:
+
+- connected (based on the connectionPolicy) according to the specified topology in the policy. Or
+- not connected (based on the connectionPolicy).
+
+If more than one matching policy is found, the priority field is used to determine which policy to use. If policies of equal priority are found
+it's assumed that these policies are to be applied in parallel.
+
+If no matching policy is found, the default cable driver and default connection policy are used. This policy can be auto-created at broker
+deploy time and should specify the cable, the connection policy (to connect or not connect nodes that don't match predefined
+policies) as well as the preferred topology. If the default policy is not to connect nodes that don't match the selector criteria, then
+a topology and default cable are not needed.
+
+> **_NOTE:_**  the `ConnectionPolicy` field can also be used to specify policies for which clusters not to connect.
+> **_NOTE:_** In the case of multiple labels being specified, the selector will need to match all the labels. Equality-based requirements will
+be supported. For e.g. labels such as `env: !production` will match any clusters that aren't labelled with `env: production`.
+
+Defining separate resources for each policy and defining a priority or precedence field, allows Submariner to prioritize the
+connectivity and cable type between a pair of clusters.
+
+The `ClusterConnectionPolicies` resources would be created and maintained on the broker cluster and synced to each participating cluster.
```

Comment:
Policies should be created locally as any other custom resource, and utilize the broker mechanism to federate them